src/components/FeaturedCard/FeaturedCard.tsx

```diff
@@ -1,16 +1,67 @@
-import {Api} from '../../apiCalls';
+import React, { Component } from "react";
+import { Api } from "../../apiCalls";
+import "./FeaturedCard.css";
 
-export const FeaturedCard = ({Api}:any) => {
-  return (
-  <div>
-    <h2>{Api.API}</h2>
-        <p>{Api.Auth}</p>
-        <p>{Api.Cors}</p>
-        <p>{Api.HTTPS}</p>
-        <p>{Api.Category}</p>
-        <p>{Api.Description}</p>
-        <p>{Api.Link}</p>
-        <textarea placeholder='Notes'></textarea>
-  </div>
-    )
-}
\ No newline at end of file
+type Props = Api & { 
+  addToFavorites: (ApiCard: Api) => void, 
+  saveNote: (ApiName: string, note: string) => void 
+};
+
+export class FeaturedCard extends Component<Props> {
+  state: Api & {note: string};
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      API: props.API,
+      Description: props.Description,
+      Auth: props.Auth,
+      HTTPS: props.HTTPS,
+      Cors: props.Cors,
+      Link: props.Link,
+      Category: props.Category,
+      note: ''
+    };
+  }
+
+  handleChange = (event: React.ChangeEvent<HTMLTextAreaElement>) => {
+    this.setState({note: event.target.value})
+  }
+
+  saveCurrentNote = () => {
+    this.props.saveNote(this.state.API, this.state.note);
+    this.setState({note: ''});
+  }
+
+  render() {
+    return (
+      <div className="featured-card">
+        <h2>{this.props.API}</h2>
+        <p>Authentication: {this.props.Auth}</p>
+        <p>Cors? {this.props.Cors}</p>
+        <p>{this.props.HTTPS}</p>
+        <p>Category: {this.props.Category}</p>
+        <p>{this.props.Description}</p>
+        <p>
+          <a href={this.props.Link} target="_blank">
+            more details
+          </a>
+        </p>
+        <textarea
+          name='note'
+          value={this.state.note}
+          className="featured-card__notes"
+          placeholder="Notes"
+          onChange={this.handleChange}
+        />
+        <button onClick={this.saveCurrentNote}>Save Notes</button>
+        <button
+          onClick={() =>
+            this.state.API && this.props.addToFavorites(this.state)
+          }
```

Comment:
Verified that the data is saving as expected to localStorage. No problems with duplicate entries either. I wonder if we can change the 'Add to Favorites' button to some kind of toggle so there's some indication that the card has been saved. Is there a way to un-favorite?